presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BooleanWriteFunction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public interface BooleanWriteFunction
```

Comment:
Should we care about type specific classes? It's JDBC, so it won't be overly fast, so boxing shouldn't be a problem. Maybe let's just have:
```
public interface WriteFunction<T>
{
    Class<?> getJavaType();

    void set(PreparedStatement statement, int index, T value)
}
```

Reply:
Or maybe make `set`:
```
void set(PreparedStatement statement, int index, Block block, int position)
```

Reply:
the `Block` version would not be applicable to predicate pushdown

Reply:
> Should we care about type specific classes? It's JDBC, so it won't be overly fast, so boxing shouldn't be a problem. Maybe let's just have:

I think we may not, but we still need `getJavaType()`, otherwise `io.prestosql.plugin.jdbc.JdbcPageSink#appendColumn` would not know how to retrieve data from a block.
And if we need `getJavaType()` and want to define these with lambdas, we need specialized interfaces (`XxxWriteFunction`)... 

Thus, we may make the `set` method generic, but this doesn't help much. (I made the change locally and it compiled OK)
OTOH, having specialized versions conveys the informations that `null` is not possible value and does not need to be handled.

For these reasons, I would prefer to retain this as it is.